massa-protocol-worker/src/protocol_worker.rs

```diff
@@ -946,8 +948,37 @@ impl ProtocolWorker {
             let mut ops = self.storage.clone_without_refs();
             ops.store_operations(new_operations.into_values().collect());
 
-            // Request propagation
-            self.propagate_ops(&ops).await;
+            // Propagate operations when their expire period isn't `max_operations_propagation_time` old.
+            let mut ops_to_propagate = ops.clone();
+            let operations_to_not_propagate = {
+                let now = MassaTime::now(0)?;
+                let read_operations = ops_to_propagate.read_operations();
+                ops_to_propagate
+                    .get_op_refs()
+                    .iter()
+                    .filter_map(|op_id| {
+                        let expire_period =
+                            read_operations.get(op_id).unwrap().content.expire_period;
+                        let expire_period_timestamp = get_block_slot_timestamp(
+                            self.config.thread_count,
+                            self.config.t0,
+                            self.config.genesis_timestamp,
+                            Slot::new(expire_period, 0),
+                        )
+                        .ok()?;
```

Comment:
don't use "?" because an attacker might choose a block slot that overflows the timestamp. If it returns Error, the op/endorsement should just NOT be propagated